tests/benchmarks/android_youtube.py

```diff
@@ -0,0 +1,189 @@
+    def setupWorkload(self):
+        # Create a results folder for each "governor/test"
+        self.out_dir = os.path.join(self.te.res_dir, governor,
+                       self.video_url.replace('/', '_'))
+        try:
+                os.stat(self.out_dir)
+        except:
+                os.makedirs(self.out_dir)
+        # Setup workload parameters
+        self.bm_params = {
+            'video_url'  : self.video_url,
+            'video_duration_s' : self.video_duration_s,
+        }
+
+    def setupGovernor(self):
+        try:
+            self.target.cpufreq.set_all_governors(self.governor);
+        except TargetError:
+            self._log.warning('Governor [%s] not available on target',
+                             self.governor)
+            raise
+
+        # Setup schedutil parameters
+        if self.governor == 'schedutil':
+            rate_limit_us = 2000
+            # Different schedutil versions have different tunables
+            tunables = self.target.cpufreq.list_governor_tunables(0)
+            if 'rate_limit_us' in tunables:
+                tunables = {'rate_limit_us' : str(rate_limit_us)}
+            else:
+                assert ('up_rate_limit_us' in tunables and
+                        'down_rate_limit_us' in tunables)
+                tunables = {
+                    'up_rate_limit_us' : str(rate_limit_us),
+                    'down_rate_limit_us' : str(rate_limit_us)
+                }
+
+            try:
+                for cpu_id in range(self.te.platform['cpus_count']):
+                    self.target.cpufreq.set_governor_tunables(
+                        cpu_id, 'schedutil', **tunables)
+            except TargetError as e:
+                self._log.warning('Failed to set schedutils parameters: {}'\
+                                 .format(e))
+                raise
+            self._log.info('Set schedutil.rate_limit_us=%d', rate_limit_us)
+
+        # Setup ondemand parameters
+        if self.governor == 'ondemand':
+            try:
+                for cpu_id in range(self.te.platform['cpus_count']):
+                    tunables = self.target.cpufreq.get_governor_tunables(cpu_id)
+                    self.target.cpufreq.set_governor_tunables(
+                        cpu_id, 'ondemand',
+                        **{'sampling_rate' : tunables['sampling_rate_min']})
+            except TargetError as e:
+                self._log.warning('Failed to set ondemand parameters: {}'\
+                                 .format(e))
+                raise
+            self._log.info('Set ondemand.sampling_rate to minimum supported')
+
+        # Report configured governor
+        governors = self.target.cpufreq.get_all_governors()
+        self._log.info('Using governors: %s', governors)
+
+# Run the benchmark in each of the supported governors
+
+video_duration_s = 60
+
+governors = [
+    'performance',
+    'ondemand',
+    'interactive',
+    'sched',
+    'schedutil',
+    'powersave',
+]
```

Comment:
Perhaps we should think about adding a new benchmark command line parameter here:
   https://github.com/ARM-software/lisa/blob/master/libs/utils/android/benchmark.py#L78
which allows to specify at command line which governors we want to test.

Let's keep for another time, but what do you think?... most of the time maybe we do not want to test with really all these governors... do you think it should be possible/useful to expose such a tunables in Jenkins?